### DISCUSSION TITLE
Connection test just shows red rectangle

### DISCUSSION THREAD

**Discussion by MatrixManAtYrService** [apache/airflow#26048] [General]
### Apache Airflow version

main (development)

### What happened

I was walking through this tutorial: https://astro-sdk-python.readthedocs.io/en/stable/getting-started/GETTING_STARTED.html which had me create an aws connection.  After doing so, I clicked the "test" button, and saw this:

<img width="403" alt="Screen Shot 2022-08-29 at 10 54 35 AM" src="https://user-images.githubusercontent.com/5834582/187253551-bbcace51-b203-444d-805e-e7608ec41b0e.png">

And this in the webserver logs:
```
[2022-08-29 10:52:59,759] {base.py:69} INFO - Using connection ID 'hSL34Jpz' for task execution.
[2022-08-29 10:52:59,759] {connection_wrapper.py:210} INFO - AWS Connection (conn_id='hSL34Jpz', conn_type='aws') credentials retrieved from login and password.
[2022-08-29 10:52:59 -0600] [64218] [WARNING] Worker with pid 64227 was terminated due to signal 11
```

### What you think should happen instead

I'm assuming that the error message was empty because the worker which was supposed to populate it died before it got a chance.  If this happens, I'd still like to get *something* in the UI. For instance, it could say: `Worker with pid 64227 was terminated due to signal 11` which would at least give me something.

### How to reproduce

test a connection.  I've also seen this with the snowflake provider, so I don't think it's aws specific

### Operating System

Mac OS

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==5.0.0
apache-airflow-providers-common-sql==1.1.0
apache-airflow-providers-ftp==3.1.0
apache-airflow-providers-http==4.0.0
apache-airflow-providers-imap==3.0.0
apache-airflow-providers-snowflake==3.2.0
apache-airflow-providers-sqlite==3.2.0

### Deployment

Virtualenv installation

### Deployment details

I've been setting up airflow venvs like this (nothing especially weird in this case)

1. clone airflow (currently at 018f1d5ebb )
2. `python scripts/ci/pre_commit/pre_commit_compile_www_assets.py`
3. run postgres in docker in a separate shell and set up the venv like so:
```
ENV_NAME="$(pwd | sed 's#/#_#g')"
pyenv virtualenv 3.10.6 $ENV_NAME
pyenv activate $ENV_NAME
export AIRFLOW_HOME=$(pwd)
pip install psycopg2
pip install ~/src/airflow
pip install 'astro-sdk-python[amazon,snowflake]>=0.11'
airflow info
sed -i .bak 's#^sql_alchemy_conn = .*$#sql_alchemy_conn = postgres://postgres:postgres@0.0.0.0#g' airflow.cfg
sed -i .bak 's#^load_examples =.*$#load_examples = False#' airflow.cfg
sed -i .bak 's#^executor =.*$#executor = LocalExecutor#' airflow.cfg
rm *.bak
airflow db init
airflow users create --username admin --password admin --firstname admin --lastname admin --role Admin --email admin@example.org
```
5. modify airflow.cfg to include the following
```
[core]
enable_xcom_pickling = True

[astro_sdk]
SQL_SCHEMA = ASTRO_SDK_SCHEMA
```
5. `airflow webserver`

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by MatrixManAtYrService** [comment]
I've been having signal 11-related problems lately (ever since I started messing around with providers).  Probably the core of this has something to do with a provider and is not a core airflow bug.  Still, I think we should be a little more verbose when it comes to communicating this error.

**Reply by aaronnewman1** [reply]
you mean, as opposed to a red rectangle?

**Comment by potiuk** [comment]
> I've been having signal 11-related problems lately (ever since I started messing around with providers). Probably the core of this has something to do with a provider and is not a core airflow bug. Still, I think we should be a little more verbose when it comes to communicating this error.

Surely. Signal 11 is SIGSEGV which generally means that there is a serious error with your installed libraries. I.e. we have completely no idea what is the error because likely your deployment is messed up (for example you have broken libraries installed, wrong binaries in your system, badly configured libc* librarires or similar).  It's generally rather bad error indicating that you have done something, airflow maintainers could not foresee and generally it crashes python interpreter, so in-general there is no way to provide any information, because at the moment it crashes, we have no way to handle it.

But If you think there is a better way and you would like to submit a fix to make a bettter message in this case - we are absolutely open to accept a fix. Since we do not know what is the root cause (somewhat broken environment in the way that crashes C-level library) you are probably one of the best people to be able to provide the fix - because you have the unique environment that generates this error. 

And since Airflow is created by > 2100 contributors - people like you who often contributed a fix when they had problem, you are most welcome to join the team.

I will convert it to discussion and if you will be able to provide a fix that will provide a better description, that would be fantastic.

**Reply by potiuk** [reply]
BTW. I recommend you to start from a clean state or restoring your OS to a clean state if you want to have Airflow working (ideally after you work out and submit the fix above to make a better description of the problem). I think there is no other easy recovery if your low-level c-libraries are messed up.

**Comment by mdcsaenz** [comment]
I also have this issue. We are on version Version: [v2.2.3](https://pypi.python.org/pypi/apache-airflow/2.2.3)
Git Version: .release:2.2.3+06c82e17e9d7ff1bf261357e84c6013ccdb3c241

**Reply by Taragolis** [reply]
If you do not use any proxy try to set environment variable `no_proxy` to `*` 

- https://github.com/apache/airflow/discussions/27351#discussioncomment-4028487
- https://github.com/apache/airflow/discussions/26977#discussioncomment-3858925
- https://github.com/python/cpython/issues/58037
- https://bugs.python.org/issue30385#msg293958
- https://wefearchange.org/2018/11/forkmacos.rst.html

**Reply by mdcsaenz** [reply]
I did a export no_proxy=* in the web pod and still get the same error. or do you mean in the airflow.cfg file?

**Reply by Taragolis** [reply]
Oh if you run in k8s that mean known macOS issues with fork process not affect to you.

Is your error somehow related to signal `11` or `SIGSEGV`?
Is it affect only to specific connection type?
Could you post logs from webserver pod after you tried test connection?

**Reply by Taragolis** [reply]
And if you also find in logs `Worker with pid XXX was terminated due to signal 11` could you also:
- set `PYTHONFAULTHANDLER` to `true`
- restart webserver pod
- try to test connection again
- post logs from webserver with traceback

**Reply by ganeshpatro321** [reply]
Hey @Taragolis 

This is the traceback for me:


```
 webserver | [2023-04-06 10:11:15,552] {connection.py:1007} INFO - Setting use_openssl_only mode to False
 webserver | Fatal Python error: Segmentation fault
 webserver | 
 webserver | Current thread 0x00000001ed988140 (most recent call first):
 webserver | File "/opt/homebrew/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 2631 in proxy_bypass_macosx_sysconf
 webserver | File "/opt/homebrew/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/urllib/request.py", line 2655 in proxy_bypass
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/vendored/requests/utils.py", line 809 in should_bypass_proxies
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/vendored/requests/utils.py", line 825 in get_environ_proxies
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/vendored/requests/sessions.py", line 759 in merge_environment_settings
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/vendored/requests/sessions.py", line 577 in request
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/network.py", line 1037 in _request_exec
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/network.py", line 856 in _request_exec_wrapper
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/network.py", line 813 in fetch
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/network.py", line 715 in _post_request
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/auth/_auth.py", line 250 in authenticate
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/connection.py", line 1125 in _authenticate
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/connection.py", line 1099 in authenticate_with_retry
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/connection.py", line 839 in __open_connection
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/connection.py", line 571 in connect
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/connection.py", line 304 in __init__
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/snowflake/connector/__init__.py", line 50 in Connect
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/airflow/providers/snowflake/hooks/snowflake.py", line 296 in get_conn
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/airflow/providers/snowflake/hooks/snowflake.py", line 365 in run
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 244 in get_first
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 512 in test_connection
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/airflow/models/connection.py", line 389 in test_connection
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/airflow/api_connexion/endpoints/connection_endpoint.py", line 193 in test_connection
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/airflow/api_connexion/security.py", line 51 in decorated
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/connexion/decorators/parameter.py", line 120 in wrapper
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/connexion/decorators/response.py", line 112 in wrapper
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/connexion/decorators/validation.py", line 196 in wrapper
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/connexion/decorators/uri_parsing.py", line 149 in wrapper
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/connexion/decorators/decorator.py", line 68 in wrapper
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/flask/app.py", line 1799 in dispatch_request
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/flask/app.py", line 1823 in full_dispatch_request
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/flask/app.py", line 2528 in wsgi_app
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/flask/app.py", line 2551 in __call__
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 179 in handle_request
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 136 in handle
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 31 in accept
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 69 in run_for_one
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/workers/sync.py", line 125 in run
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/workers/base.py", line 142 in init_process
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 589 in spawn_worker
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 622 in spawn_workers
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 551 in manage_workers
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 202 in run
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/app/base.py", line 72 in run
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/app/base.py", line 231 in run
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/app/wsgiapp.py", line 67 in run
 webserver | File "/Users/ganeshpatro/workstation/code/airflow/venv/lib/python3.8/site-packages/gunicorn/__main__.py", line 7 in <module>
 webserver | File "/opt/homebrew/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 87 in _run_code
 webserver | File "/opt/homebrew/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 194 in _run_module_as_main
 webserver | [2023-04-06 10:11:15 +0530] [55642] [WARNING] Worker with pid 55676 was terminated due to signal 11
```

**Reply by potiuk** [reply]
Yes. Segmentation fault is the sign that this is MacOS ARM problem.  Similar Issues https://github.com/apache/airflow/discussions?discussions_q=Negsignal.SIGSEGV  -> you will find more details and possible workarounds there. 

Setting environment variable NO_PROXY='*' helps. You just have to make sure it is set and exported in your POD. You likely will need to build your own image to set it, it's not enough that you enter the container and set it - it has to be set for the whole image. One of the ways to do that is to extend the image.

You can see more description about buildin your image and extending the base one here https://airflow.apache.org/docs/docker-stack/build.html#examples-of-image-extending (you likely need to use ENV command). And then you need to use the image.

Let us know if it worked for you.
